mlflow/cli.py

```diff
@@ -181,6 +182,15 @@ def ui(backend_store_uri, default_artifact_root, host, port, gunicorn_opts):
         else:
             default_artifact_root = DEFAULT_LOCAL_FILE_AND_ARTIFACT_PATH
 
+    try:
+        _get_store(backend_store_uri)
+    except Exception as e:  # pylint: disable=broad-except
+        # We need a broad exception here to catch exceptions in optional
+        # dependencies like SQLAlchemy
+        _logger.error("Error related to option --backend-store-uri")
```

Comment:
I'm not sure if it's 100% sure it's related to backend-store-uri. Maybe just "Error initializing backend store"?

Btw, `ARTIFACT_ROOT_ENV_VAR` is not passed here and defaults to None. Could this cause problems? e.g., if some validation checks for it? Maybe we should also pass that.